### PR TITLE
Add tests for Zero Non Deal Bids Warning Only in Debug

### PR DESCRIPTION
**Pull request by osulzhenko**
### 🔧 Type of changes
- [ ] new bid adapter
- [ ] update bid adapter
- [ ] new feature
- [ ] new analytics adapter
- [ ] new module
- [ ] bugfix
- [ ] documentation
- [ ] configuration
- [x] tech debt (test coverage, refactorings, etc.)

### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🔎 New Bid Adapter Checklist
- [ ] verify email contact works
- [ ] NO fully dynamic hosts
- [ ] geographic host parameters are NOT required
- [ ] NO direct use of HTTP is prohibited - *implement an existing Bidder interface that will do all the job*
- [ ] if the ORTB is just forwarded to the endpoint, use the generic adapter - *define the new adapter as the alias of the generic adapter*
- [ ] cover an adapter configuration with an integration test

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 🏎 Quality check

- [ ] Are your changes following [our code style guidelines](https://github.com/prebid/prebid-server-java/blob/master/docs/developers/code-style.md)?
- [ ] Are there any breaking changes in your code?
- [ ] Does your test coverage exceed 90%?
- [ ] Are there any erroneous console logs, debuggers or leftover code in your changes?

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 1]
```diff
@@ -113,8 +113,8 @@ class BidValidationSpec extends BaseSpec {
 
         where:
         bidRequest << [BidRequest.getDefaultBidRequest(DistributionChannel.APP).tap {
-                           it.dooh = Dooh.defaultDooh
-                       },
+            it.dooh = Dooh.defaultDooh
+        },
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:117]: Update style please

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 2]
```diff
@@ -247,6 +254,46 @@ class BidValidationSpec extends BaseSpec {
...
+        1     | 0    | 0                             | null
+        1     | 0    | null                          | PBSUtils.randomNumber
+        1     | 0    | null                          | null
+    }
+
+    def "PBS should drop invalid bid without debug error when request debug disabled and bid price is #bidPrice and deal id is #dealId"() {
+        given: "Default basic BidRequest with generic bidder"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            test = 0
+            ext.prebid.debug = 0
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:275]: If you don't mind, please make test and debug like enum DISABLED(0), ENABLED(1)

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 3]
```diff
@@ -247,6 +254,46 @@ class BidValidationSpec extends BaseSpec {
...
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Invalid bid should be deleted"
+        assert response.seatbid.size() == 0
+
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:292]: Strangely, seat bid is not about invalid bids, better to check the seatNonBid field.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 4]
```diff
@@ -247,6 +254,46 @@ class BidValidationSpec extends BaseSpec {
...
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            test = 0
+            ext.prebid.debug = 0
+        }
+
+        and: "Bid response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+        def bid = bidResponse.seatbid.first().bid.first()
+        bid.dealid = dealId
+        bid.price = bidPrice
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:282]:
```
   def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap{
     it.bid.tap{
     dealid = dealId
     price = bidPrice
            }   
   }
```

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 5]
```diff
@@ -247,6 +254,46 @@ class BidValidationSpec extends BaseSpec {
...
+        0     | 1    | null                          | PBSUtils.randomNumber
+        0     | 1    | null                          | null
+        1     | 0    | PBSUtils.randomNegativeNumber | null
+        1     | 0    | PBSUtils.randomNegativeNumber | PBSUtils.randomNumber
+        1     | 0    | 0                             | null
+        1     | 0    | null                          | PBSUtils.randomNumber
+        1     | 0    | null                          | null
+    }
+
+    def "PBS should drop invalid bid without debug error when request debug disabled and bid price is #bidPrice and deal id is #dealId"() {
```
Comment (marki1an) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:271]: Did the bidder call happen in such case?

**Review by marki1an — CHANGES_REQUESTED** [review]

**Review by marki1an — APPROVED** [review]
